x-pack/plugins/security_solution/public/cloud_security_posture/components/entity_insight.tsx

```diff
@@ -60,6 +65,39 @@ export const EntityInsight = <T,>({
 
   const isVulnerabilitiesFindingForHost = hasVulnerabilitiesFindings && fieldName === 'host.name';
 
+  const { signalIndexName } = useSignalIndex();
+
+  const entityFilter = useMemo(() => ({ field: fieldName, value: name }), [fieldName, name]);
+
+  const { to, from } = useGlobalTime();
+
+  const { items: alertsData } = useAlertsByStatus({
+    entityFilter,
+    signalIndexName,
+    queryId: DETECTION_RESPONSE_ALERTS_BY_STATUS_ID,
+    to,
+    from,
+  });
+
+  const alertsOpenCount = alertsData?.open?.total || 0;
+
+  const alertsAcknowledgedCount = alertsData?.acknowledged?.total || 0;
+
+  const alertsCount = alertsOpenCount + alertsAcknowledgedCount;
+
+  if (alertsCount > 0) {
+    insightContent.push(
+      <>
+        <AlertsPreview
+          alertsData={alertsData}
```

Comment:
passing the whole alertsData object forces us to have the logic of filtering in only `open` and `acknowledged` in two places, here and in the `AlertsPreview` component. Let's make AlertsPreview pure from this logic. You can filter in only `open` and `acknowledged` here, then calculate the total by iterating over the keys of `alertsData` and summing up the total.